fix: #190 gradient color stop offfset computation

fidlip opened the pull request:
Fix for issue #190: Linear gradient stops are badly interpreted

fidlip committed:
fix: #190 gradient color stop offfset computation

felixfbecker commented on src/gradients.ts:
```diff
@@ -51,7 +51,7 @@ export function convertLinearGradient(
 
 	const getColorStops = (colorStop: gradientParser.ColorStop, index: number): SVGStopElement => {
-		const offset = `${(index / (colorStops.length - 1)) * 100}%`
+	        const offset = `${colorStop.length.value}${colorStop.length.type}`;
```
I don't think this is correct SVG, according to MDN the `offset` attribute accepts either a number or a percentage. This would be valid only when `colorStop.length.type` is `%`, but it could be `px` or `em` which would not be valid to be added to the `offset` attribute.

fidlip replied:
You are right, I tried to fix it in next commit. Thank you.

fidlip committed:
Fixed linear-gradient stops computation to folow MDN.

Interprets percentage value in linear-gradient stops like it should be and leaves original solution for other units as fallback. We do not know size of filled area so can not convert values to exact percentage or number representation.

fidlip commented:
Hi @felixfbecker, It should be fixed now. Is it OK from your point of view?
